api/src/main/java/org/apache/cloudstack/api/command/user/vm/AddIpToVmNicCmd.java

```diff
@@ -169,17 +161,25 @@ public long getEntityOwnerId() {
 
     @Override
     public void create() throws ResourceAllocationException {
-        String ip;
         NicSecondaryIp result;
-        String secondaryIp = null;
-        if ((ip = getIpaddress()) != null) {
-            if (!NetUtils.isValidIp4(ip)) {
-                throw new ServerApiException(ApiErrorCode.INTERNAL_ERROR, "Invalid ip address " + ip);
+        boolean isIpv4 = true;
```

Comment:
What about moving this validation to a "service"  layer? This is the "view/controller" part of the design. It should only receive the request, get the data and dispatch to the service layer where the business gets done.

Reply:
Thanks, @rafaelweingartner. Updated it.